src/Common/Core/Impl/Extensions/ArraySegmentExtensions.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.Common.Core {
+    public static class ArraySegmentExtensions {
+        public static async Task<byte[]> ToByteArrayAsync(this ArraySegment<byte> source, int count) {
```

Comment:
`ArraySegment` has its own count - if this is a generic helper now, it should probably offer an overload that defaults `count` to that.